app/register/page.tsx

```diff
@@ -1,34 +1,55 @@
+'use client';
+
 import Link from 'next/link';
 import { redirect } from 'next/navigation';
-import { createUser, getUser } from '@/lib/db';
+import { createUserFromRegister, getUserFromRegister } from '@/lib/actions';
 import { SimpleLayout } from '@/components/page-layouts';
 import { Label } from '@/components/ui/label';
 import { Input } from '@/components/ui/input';
 import { Button } from '@/components/ui/button';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
 
 export default function Page() {
 
-  async function register(formData: FormData) {
-    'use server';
-    let name = formData.get('name') as string;
-    let phone = formData.get('phone') as string;
-    let email = formData.get('email') as string;
-    let password = formData.get('password') as string;
-    let confirmPassword = formData.get('confirm-password') as string;
-    
-    // TODO do confirm password checking locally with react-hook-forms
-    if (confirmPassword !== password) {
-      return 'Passwords do not match';
-    }
+  const schema = z.object({
+    name: z.string().trim().min(1, {message: "Name is required"}),
+    phone: z.string().min(5, {message: "Phone must contain at least 5 characters"}).max(32, {message: "Phone cannot contain more than 32 characters"}).regex(/^\+?[0-9\s-]+$/, {message: "Phone contains invalid characters"}).optional().or(z.literal("")),
+    email: z.string().trim().min(1, {message: "Email is required"}).email(),
+    password: z.string().min(8, {message: "Password must contain at least 8 characters"}),
+    confirmPassword: z.string().min(8, {message: "Password must contain at least 8 characters"}),
+  }).refine(schema => schema.confirmPassword === schema.password, {
+      message: "Oops! Passwords don't match. Try again.",
+      path: ["confirmPassword"],
+    });
+
+  type FormFields = z.infer<typeof schema>;
 
-    let user = await getUser(email);
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors },
+  } = useForm<FormFields>({
+    resolver: zodResolver(schema),
+  });
 
-    if (user.length > 0) {
-      // TODO: Handle errors with useFormStatus
-      return 'User already exists';
-    } else {
-      await createUser(name, phone, email, password);
+  const onSubmit: SubmitHandler<FormFields> = async (data) => {
+    
+    try {
+      let user = await getUserFromRegister(data.email);
```

Comment:
We don't want to do this validation client side, this way someone can call createUserFromRegister(data) directly and change an existing users email

Fix you one!

Reply:
Checks should always be done serverside in actions.ts. client side checks are for UX, making it easier to use. Server side checks are for security and business logic

Reply:
Fixed - let me know if it's alright now?